internal/processors/enrichevents/enrichevents.go

```diff
@@ -130,17 +108,35 @@ func (v *processor) validateEvent(event *cloudevent.RawEvent) error {
 	return nil
 }
 
-func (v *processor) validateEventQueryFields(event EventData) error {
+func (v *processor) validateAndFormatEvent(header cloudevent.CloudEventHeader, event EventData, storedEvtObj *occurrences.Event) error {
 	if event.Duration != nil {
-		if _, err := time.ParseDuration(*event.Duration); err != nil {
-			return fmt.Errorf("invalid %s event duration: %w", event.Name, err)
+		dur, err := time.ParseDuration(*event.Duration)
+		if err != nil {
+			return fmt.Errorf("invalid duration for event %q: %w", event.Name, err)
 		}
+		storedEvtObj.EventDuration = dur
 	}
 
 	if event.Time != nil {
-		if _, err := time.Parse(time.RFC3339, *event.Time); err != nil {
-			return fmt.Errorf("invalid %s event duration: %w", event.Name, err)
+		t, err := time.Parse(time.RFC3339, *event.Time)
+		if err != nil {
+			return fmt.Errorf("invalid time for event %q: %w", event.Name, err)
+		}
+		storedEvtObj.EventTime = t
+	}
+
+	storedEvtObj.CloudEventID = header.ID
+	storedEvtObj.Subject = header.Subject
+	storedEvtObj.Source = header.Source
+	storedEvtObj.Producer = header.Producer
+	storedEvtObj.EventName = event.Name
+
+	if len(event.Metadata) > 0 {
+		var tmp map[string]interface{}
```

Comment:
Maybe it's not an object? We said "free-form" in the migration.

There's a [`Valid` function in `encoding/json`](https://pkg.go.dev/encoding/json#Valid), maybe that?